mitosheet/src/components/elements/Toggle.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mito
+import React from 'react';
+
+// import css
+import '../../../css/elements/Toggle.css'
+import { classNames } from '../../utils/classNames';
+
+interface ToggleProps {
+    /** 
+        * @param value - The state of the toggle. True (On) or False (Off)
+    */
+    value: boolean;
+
+    /**
+        * @param onChange - Handles actually changing the value of the filter 
+    */
+    onChange: () => void;
+
+    /**
+        * @param [disabled] - Disables the toggle button
+    */
+    disabled?: boolean
+}
+
+/**
+ * The Toggle component. If open, the select dropdown automatically 
+ * closes when the user clicks.
+ * 
+ * The Toggle component utilizes the html input element with a type checkbox, so that we 
+ * follow the design patterns of html elements. In order to create our custom designed toggle, 
+ * we hide the default html checkbox and style the label property. 
+ * 
+ * Inspired by: https://stackoverflow.com/questions/39846282/how-to-add-the-text-on-and-off-to-toggle-button and 
+ * https://www.youtube.com/watch?v=N8BZvfRD_eU
```

Comment:
Otherwise, this is a great doc comment. Thanks for this - really helpful to understand what/why is happening here!